lib/src/main/java/com/cloudinary/android/uploadwidget/ui/ThumbnailsAdapter.java

```diff
@@ -59,16 +61,35 @@ public void onClick(View v) {
         });
 
         Context context = holder.itemView.getContext();
-        float thumbnailSize = context.getResources().getDimension(R.dimen.thumbnail_size);
-        BitmapManager.get().load(context, uri, (int) thumbnailSize, (int) thumbnailSize, new BitmapManager.LoadCallback() {
-            @Override
-            public void onSuccess(Bitmap bitmap, Dimensions originalDimensions) {
-                holder.imageView.setImageBitmap(bitmap);
-            }
-
-            @Override
-            public void onFailure() { }
-        });
+        int thumbnailSize = (int) context.getResources().getDimension(R.dimen.thumbnail_size);
+
+        MediaType mediaType = UriUtils.getMediaType(context, uri);
+        if (mediaType == MediaType.IMAGE) {
+            holder.mediaTypeIcon.setVisibility(View.GONE);
+
+            BitmapManager.get().load(context, uri, thumbnailSize, thumbnailSize, new BitmapManager.LoadCallback() {
+                @Override
+                public void onSuccess(Bitmap bitmap, Dimensions originalDimensions) {
+                    holder.imageView.setImageBitmap(bitmap);
```

Comment:
Can't these async calls scramble the results? If the user scrolls while the async tasks is finished, this specific holder may already be bound the another data item, we need to add some verification that we're still dealing with the same position before setting the image.